core/server/i18n.js

```diff
@@ -82,22 +98,55 @@ I18n = {
      *  - Polyfill node.js if it does not have Intl support or support for a particular locale
      */
     init: function init() {
-        // read file for current locale and keep its content in memory
-        blos = fs.readFileSync(__dirname + '/translations/' + currentLocale + '.json');
-
-        // if translation file is not valid, you will see an error
-        try {
-            blos = JSON.parse(blos);
-        } catch (err) {
-            blos = undefined;
-            throw err;
+        // i18n.init() is only called: during Ghost's initialization twice, and when the translation files are not in memory;
+        // for the first call, settings for language and theme are not yet available, and English default is used for core;
+        // settings are available for the second initialization call, allowing full internationalization of core and theme
+        // (see file core/server/index.js for Ghost's initialization, and function i18n.findString above)
+
+        var blosCore, blosTheme, hasBuiltInLocaleData, IntlPolyfill;
+
+        currentLocale = settingsCache.get('default_lang') || 'en';
+        activeTheme = settingsCache.get('active_theme') || '';
+
+        // Reading files for current locale and active theme and keeping their content in memory
+        // (during Ghost's initialization, when English is the language in settings, the second call
+        // to i18n.init() doesn't read again the default English file for core if already in memory).
+        if (blos === undefined || currentLocale !== 'en') {
+            blosCore = fs.readFileSync(__dirname + '/translations/' + currentLocale + '.json');
```

Comment:
There is some protection in i18n.init for missing translation files in non-i18n themes, but not yet for core translation file, that for English is already there (currently en.json). With the new setting, surely protection should be added for wrong locales.